Microsoft.Toolkit.Uwp.UI/Cache/CacheBase.cs

```diff
@@ -383,9 +434,27 @@ private static ulong CreateHash64(string str)
             if (baseFile == null || await IsFileOutOfDateAsync(baseFile, CacheDuration).ConfigureAwait(MaintainContext))
             {
                 baseFile = await folder.CreateFileAsync(fileName, CreationCollisionOption.ReplaceExisting).AsTask().ConfigureAwait(MaintainContext);
+
+                uint retries = 0;
                 try
                 {
-                    instance = await DownloadFileAsync(uri, baseFile, preCacheOnly, cancellationToken, initializerKeyValues).ConfigureAwait(false);
+                    while (retries < RetryCount)
+                    {
+                        try
+                        {
+                            instance = await DownloadFileAsync(uri, baseFile, preCacheOnly, cancellationToken, initializerKeyValues).ConfigureAwait(MaintainContext);
+
+                            if (instance != null)
+                            {
+                                break;
+                            }
+                        }
+                        catch (FileNotFoundException)
```

Comment:
I have seen requests finish with no data.. Hence retry loop.. This happened on very rare occasions before as well but wasn't very noticable.. Swallowing only FileNotFound that is throw by InitializeAsync btw